adapters/handlers/grpc/v1/parse_search_request.go

```diff
@@ -364,46 +365,104 @@ func extractGroupBy(groupIn *pb.GroupBy, out *dto.GetParams) (*searchparams.Grou
 	var targetVectors *[]string
+	var targets *pb.Targets
+
+	extract := func(targets *pb.Targets, targetVectors *[]string) (*[]string, *pb.Targets) {
+		if targets != nil {
+			return &targets.Targets, targets
+		} else {
+			return targetVectors, nil
+		}
+	}
 	if hs := req.HybridSearch; hs != nil {
-		targetVectors = &hs.TargetVectors
+		targetVectors, targets = extract(hs.Targets, &hs.TargetVectors)
 	}
 	if na := req.NearAudio; na != nil {
-		targetVectors = &na.TargetVectors
+		targetVectors, targets = extract(na.Targets, &na.TargetVectors)
 	}
 	if nd := req.NearDepth; nd != nil {
-		targetVectors = &nd.TargetVectors
+		targetVectors, targets = extract(nd.Targets, &nd.TargetVectors)
 	}
 	if ni := req.NearImage; ni != nil {
-		targetVectors = &ni.TargetVectors
+		targetVectors, targets = extract(ni.Targets, &ni.TargetVectors)
 	}
 	if ni := req.NearImu; ni != nil {
-		targetVectors = &ni.TargetVectors
+		targetVectors, targets = extract(ni.Targets, &ni.TargetVectors)
 	}
 	if no := req.NearObject; no != nil {
-		targetVectors = &no.TargetVectors
+		targetVectors, targets = extract(no.Targets, &no.TargetVectors)
 	}
 	if nt := req.NearText; nt != nil {
-		targetVectors = &nt.TargetVectors
+		targetVectors, targets = extract(nt.Targets, &nt.TargetVectors)
 	}
 	if nt := req.NearThermal; nt != nil {
-		targetVectors = &nt.TargetVectors
+		targetVectors, targets = extract(nt.Targets, &nt.TargetVectors)
 	}
 	if nv := req.NearVector; nv != nil {
-		targetVectors = &nv.TargetVectors
+		targetVectors, targets = extract(nv.Targets, &nv.TargetVectors)
 	}
 	if nv := req.NearVideo; nv != nil {
-		targetVectors = &nv.TargetVectors
+		targetVectors, targets = extract(nv.Targets, &nv.TargetVectors)
+	}
+
+	var combination *dto.TargetCombination
+	if targets != nil {
+		var err error
+		if combination, err = extractTargets(targets); err != nil {
+			return nil, nil, err
+		}
+	} else if targetVectors != nil && len(*targetVectors) > 1 {
+		// here weights need to be added if the default combination requires it
+		combination = &dto.TargetCombination{Type: dto.DefaultTargetCombinationType}
 	}
 
 	if targetVectors != nil && len(*targetVectors) == 0 && len(class.VectorConfig) > 1 {
-		return nil, fmt.Errorf("class %s has multiple vectors, but no target vectors were provided", class.Class)
+		return nil, nil, fmt.Errorf("class %s has multiple vectors, but no target vectors were provided", class.Class)
+	}
+	return targetVectors, combination, nil
+}
+
+func extractTargets(in *pb.Targets) (*dto.TargetCombination, error) {
+	if in == nil {
+		return nil, nil
 	}
-	if targetVectors != nil && len(*targetVectors) > 1 {
-		return nil, fmt.Errorf("cannot provide multiple target vectors when searching, only one is allowed")
+
+	var combinationType dto.TargetCombinationType
+	weights := make(map[string]float32, len(in.Weights))
+	if in.Combination == pb.CombinationMethod_COMBINATION_METHOD_TYPE_AVERAGE {
```

Comment:
you can consider using `switch` statement here for better readability and maintainability